docs/spec/spec.md

```diff
@@ -3252,3 +3264,127 @@ type Product record {
+
+The user is responsible for implementing the logic to collect the keys of the data to be loaded into the `DataLoader` in the `loadXxx` method. The GraphQL Engine guarantees the execution of the `loadXxx` method prior to the `xxx` method. Subsequently, the user can implement the logic to retrieve the data from the `DataLoader` within the `xxx` method.
+
+#### 11.1.2.2. The `@dataloader:Loader` Annotation
+
+The `@dataloader:Loader` annotation is used to configure the `loadXxx` method. It serves the purpose of avoiding the addition of `loadXxx` as a field in the GraphQL schema. This annotation accepts a map of batch load functions to be used by the `DataLoader`.
+
+The map consists of unique identifiers as keys and user-written batch load functions as values. Each unique key represents a separate `DataLoader` object. When the GraphQL Engine encounters the `@dataloader:Loader` annotation, it uses the provided configuration to create `DataLoader` objects for each unique key.
+
+Using the same unique key across multiple `@dataloader:Loader` annotations will result in the same `DataLoader` object being shared by multiple GraphQL fields. This enables efficient batching and caching of data loading operations. The following is the type definition of this annotation.
+
+```ballerina
+# Provides configurations for the DataLoaders.
+# + batchFunctions - The map of batch function to be used in the DataLoader
+public type LoaderConfig record {|
+    map<(isolated function (readonly & anydata[] keys) returns anydata[]|error)> batchFunctions;
+|};
+
+# The annotation to configure the load method with batch functions.
+public annotation LoaderConfig Loader on object function;
+```
+
+#### 11.1.2.3. The Batch Load Function
+
+In the context of the `@dataloader:Loader` annotation, the `batchFunctions` map consists of batch load functions associated with the unique key of the DataLoader. These batch load functions are responsible for retrieving data based on an array of keys and returning an array of corresponding results or an `error` if the operation fails.
+
+>**Note:** It is important to ensure that the batch load function returns an array of results that matches the length of the keys array provided as input. If the lengths do not match, the DataLoader will return an error when the `get` method is called.
```

Comment:
This shouldn't be a note. It is a key part of the functionality.

Reply:
Fixed with [3a43a46](https://github.com/ballerina-platform/module-ballerina-graphql/pull/1505/commits/3a43a46cb42516d27623aa7bd6b535a4b90c9bb5)